src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {

```

Comment:
All I'm saying is that getDeclarationFlags will return StatementFlags and it just seems simpler to pass in the appropriate StatementFlags mask to test against. parseModuleElementOfKind is only called from parseStatement, parseModuleElement, and parseSourceElement, and those are the functions that anything else should call.